docs/distribution/nightly.md

```diff
@@ -0,0 +1,82 @@
+---
+layout: developer-doc
+title: Nightly Builds
+category: distribution
+tags: [distribution, release, nightly]
+order: 10
+---
+
+# Nightly Builds
+
+This document describes the infrastructure for automated nightly builds.
+
+<!-- MarkdownTOC levels="2,3" autolink="true" -->
+
+- [Triggering the Build](#triggering-the-build)
+- [Nightly Build Versions](#nightly-build-versions)
+- [Release Notes](#release-notes)
+
+<!-- /MarkdownTOC -->
+
+## Triggering the Build
+
+The build may be triggered on two possible events:
+
+- automatically, at 4am UTC after each working day (that is, on Tuesday to
+  Saturday),
+- manually, if a commit with message containing `[release: nightly]` is pushed
+  to the `main` branch.
```

Comment:
How strict is this trigger? Will it work if someone accidentally forgets the space or adds too many spaces?

Reply:
It is strict, looks for the exact substring.

Do you think it should be more lenient? However it may be problmatic to implement because the check is performed in the wrofklow before it starts so we can only use the [GitHub expression syntax](https://docs.github.com/en/actions/reference/context-and-expression-syntax-for-github-actions#functions) and I think the only relevant function that we have there, is `contains`.

Reply:
I thought it supported some rudimentary globbing?